providers/task.rb

```diff
@@ -45,6 +45,8 @@
     options['RL'] = 'HIGHEST' if @new_resource.run_level == :highest
     options['IT'] = '' if @new_resource.interactive_enabled
     options['D'] = @new_resource.day if @new_resource.day
+    options['EC'] = "\"#{@new_resource.channel_name}\" " unless @new_resource.channel_name.nil?
+  	options['MO'] = "\"#{@new_resource.event_modifier}\" " unless @new_resource.event_modifier.nil?
```

Comment:
```options['MO']``` is already being computed above.  We should probably add ```:on_event``` as a new ```frequency``` attribute that affects the schedule and perform validation here to ensure that we aren't providing any other flags that conflict with ```:on_event``` and further check that EC isn't provided when frequency is anything but ```:on_event```.